xbmc/windows/GUIMediaWindow.cpp

```diff
@@ -770,8 +770,22 @@ bool CGUIMediaWindow::Update(const std::string &strDirectory, bool updateFilterP
   }
 
   if (m_vecItems->GetLabel().empty())
+  {
     m_vecItems->SetLabel(CUtil::GetTitleFromPath(m_vecItems->GetPath(), true));
 
+    // Removable sources
```

Comment:
No, GetTitleFromPath will always returns a cleaned up path suitable as a label.
But maybe you mean the removables should be checked first, and GetTitleFromPath only called if empty, and then I agree ;)